src/actors/sender_can.rs

```diff
@@ -1,46 +1,44 @@
-use tokio::sync::mpsc;
-//use futures_timer::Delay;
-//use std::time::Duration;
-use tokio_socketcan::CANFrame;
-use super::can_handler;
+//! Sender CAN actor
+//!
+//! Actor responsible for sending can messages into a can socket
 
+use tokio::sync::{mpsc};
+
+use crate::util::canutil;
+use crate::util::canutil::{CANFrame, CANSocket};
+
+/// Messages that the Actor Can Receive
 enum SenderCANMessages {
     SendToID {
-        id: Option<String>,
-        message: Option<String>,
-        cycle_time: Option<String>,

```

Comment:
Since this comes from stdin doesn't everything come as a string? Won't it generate an error to parse a sting as an int?

Reply:
From my point of view the sender can actor should receive a message that is prepared for him to just send

Other actors should be the one responsible for creating the message with the necessary types
For instance STDIN actor should be the one responsible for parsing the input strings into numbers that the sender actor understands directly

That's what I was going for in here. What do you think ? @FranciscoProspero

Reply:
I agree. It makes sense for the send actor the receive already the correct information.